hooks/deploy-product-onexit.sh

```diff
@@ -100,4 +100,68 @@ else
     echo "INFO Successfully deployed manifests for v1.32"
 fi
 
+echo "INFO Checking current k8s-primary-cni value in BSS"
+CNI_VALUE=$(cray bss bootparameters list --hosts Global --format json | jq -r '.[]."cloud-init"."meta-data"."k8s-primary-cni"')
+
+if [[ "$CNI_VALUE" == "cilium" ]]; then
+    echo "INFO k8s-primary-cni is already set to 'cilium'. Skipping migration workflow."
+else
+    echo "INFO k8s-primary-cni is '$CNI_VALUE'. Proceeding with migration workflow."
+
+    echo "INFO Generating Cilium workflow manifest"
+
+    if [[ ! -f /usr/share/doc/csm/workflows/cilium/generateCiliumLiveMigration.py ]]; then
+        echo "ERROR Missing file: generateCiliumLiveMigration.py"
+        exit 1
+    fi
+
+    /usr/share/doc/csm/workflows/cilium/generateCiliumLiveMigration.py
+    if [[ $? -ne 0 ]]; then
+        echo "ERROR Failed to generate Cilium workflow manifest"
+        exit 1
+    fi
+
+    if [[ ! -f /usr/share/doc/csm/workflows/cilium/cilium-live-migration.yaml ]]; then
+        echo "ERROR Missing file: cilium-live-migration.yaml"
+        exit 1
+    fi
+
+    echo "INFO Applying Cilium workflow manifest"
+    kubectl apply -f /usr/share/doc/csm/workflows/cilium/cilium-live-migration.yaml -n argo
+    if [[ $? -ne 0 ]]; then
```

Comment:
Combine into a single line:
```suggestion
    if ! kubectl apply -f /usr/share/doc/csm/workflows/cilium/cilium-live-migration.yaml -n argo; then
```
Same thing, if you want to leave it as is to match the rest in this file, that's okay.